tpu-client-next/src/connection_workers_scheduler.rs

```diff
@@ -91,6 +92,72 @@ pub struct ConnectionWorkersSchedulerConfig {
 }
 
 impl ConnectionWorkersScheduler {
+    pub async fn run(
+        config: ConnectionWorkersSchedulerConfig,
+        leader_updater: Box<dyn LeaderUpdater>,
+        transaction_receiver: mpsc::Receiver<TransactionBatch>,
+        cancel: CancellationToken,
+    ) -> Result<SendTransactionStatsPerAddr, ConnectionWorkersSchedulerError> {
+        <ConnectionWorkersScheduler as ConnectionWorkersSchedulerImpl>::run(
+            config,
+            leader_updater,
+            transaction_receiver,
+            cancel,
+        )
+        .await
+    }
+}
+
+#[async_trait]
+impl ConnectionWorkersSchedulerImpl for ConnectionWorkersScheduler {
+    async fn send_to_workers(
+        workers: &mut WorkersCache,
+        leaders: &[SocketAddr],
+        transaction_batch: TransactionBatch,
+    ) -> Result<(), ConnectionWorkersSchedulerError> {
```

Comment:
Consider moving this into a "runtime configuration", via an argument to `run()`, rather than a compile time configuration.
As we discussed in a call.